content/courses/advanced-ada/chapters/expressions.rst

```diff
@@ -3,6 +3,391 @@ Expressions
 
 .. include:: ../../global.txt
 
+Expressions: Definition
+-----------------------
+
+According to the Ada Reference Manual, an expression "is a formula that defines
+the computation or retrieval of a value." Also, when an expression is evaluated,
+the computed or retrieved value always has an associated type.
+
+Even though the definition above is very simple, Ada expressions are actually
+very flexible |mdash| and they can also be very complex. In fact, if you read
+the :arm:`corresponding section <4-4>` of the Ada Reference Manual, you'll
+quickly discover that they include elements such as relations, membership
+choices, terms and primaries. In this section, we present examples of just some
```

Comment:
Since relations, terms, factors, and primaries are classic expression elements, so as an aside you may want to differentiate those from, for example, membership tests and the other elements that are unique to Ada. I'm thinking just something like "Some of these are classic elements of expressions in programming languages, although some of their forms are unique to Ada."

Reply:
Good point, thanks! I'm taking it verbatim.